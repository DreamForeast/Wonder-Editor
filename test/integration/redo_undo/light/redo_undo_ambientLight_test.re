open Wonder_jest;

open Expect;

open Expect.Operators;

open Sinon;

let _ =
  describe("redo_undo: ambientLight", () => {
    let sandbox = getSandboxDefaultVal();

    let _changeColorAndPushUndoStack = value => {
      let sourceColor = HeaderSettingTool.Scene.Ambient.getColor();

      HeaderSettingTool.Scene.Ambient.changeColor(value);

      HeaderSettingTool.Scene.Ambient.closeColorPicker(~color=sourceColor, ());
    };

    let _simulateTwiceChangeAmbientLight = () => {
      BuildCanvasTool.buildFakeCanvas(sandbox);

      let color1 = PickColorTool.buildColor1();
      let color2 = PickColorTool.buildColor2();

      _changeColorAndPushUndoStack(color1);
      _changeColorAndPushUndoStack(color2);
    };

    let _beforeEach = () =>
      MainEditorSceneTool.createDefaultSceneAndNotInit(sandbox);

    beforeEach(() => {
      sandbox := createSandbox();

      MainEditorSceneTool.initStateWithJob(
        ~sandbox,
        ~isInitJob=false,
        ~noWorkerJobRecord=
          NoWorkerJobConfigToolEngine.buildNoWorkerJobConfig(
            ~initPipelines=
              {|
               [
           {
             "name": "default",
             "jobs": [
               {
                 "name": "init_transform_gizmos"
               }
             ]
           }
         ]
               |},
            (),
          ),
        (),
      );

      MainEditorSceneTool.prepareGl(sandbox);

      StateLogicService.getAndSetEngineState(MainUtils._handleEngineState);

      EventListenerTool.buildFakeDom()
      |> EventListenerTool.stubGetElementByIdReturnFakeDom;
    });
    afterEach(() => restoreSandbox(refJsObjToSandbox(sandbox^)));

    RedoUndoTool.testRedoUndoTwoStep(
      sandbox,
      "prepare first step: set currentSceneTreeNode",
      (_simulateTwiceChangeAmbientLight, _beforeEach, () => ()),
      ()
      /* BuildComponentTool.buildHeader(TestTool.buildEmptyAppState()) */
      => HeaderSettingTool.UI.buildSetting(~isShowSceneModal=true, ()));
  });