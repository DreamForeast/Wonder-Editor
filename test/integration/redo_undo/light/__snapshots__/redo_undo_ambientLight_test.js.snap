// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redo_undo: ambientLight prepare first step: set currentSceneTreeNode test redo operate test redo one step undo step which from second to zero, redo step which from zero to first 1`] = `
<div
  className="header-item"
>
  <div
    className="component-item"
  >
    <span
      className="item-title"
      onClick={[Function]}
      onMouseOver={[Function]}
    >
      Setting
    </span>
  </div>
  <article
    className="wonder-settingScene-modal"
  >
    <div
      className="modal-item"
    >
      <div
        className="modal-item-header"
      >
        Scene
        <img
          onClick={[Function]}
          src="./public/img/close.png"
        />
      </div>
      <div
        className="modal-item-content"
      >
        <div
          className="setting-scene-ambient"
        >
          <div
            className="content-field"
          >
            <article
              className="color-pick"
            >
              <div
                className="item-header"
                title="set ambient color which is global."
              >
                AmbientColor
              </div>
              <div
                className="item-content item-color"
              >
                <div
                  className="color-hex"
                  onClick={[Function]}
                  style={
                    Object {
                      "background": "#7df1e8",
                    }
                  }
                />
                <div
                  className="color-select"
                  onClick={[Function]}
                >
                  <img
                    src="./public/img/color.png"
                  />
                </div>
              </div>
            </article>
          </div>
        </div>
        <div
          className="setting-scene-skybox"
        >
          <div
            className="item-header"
            title="set scene skybox->cubemap"
          >
            Skybox Cubemap
          </div>
          <div
            className="item-content"
          >
            <div
              className="cubemap-img"
              onClick={[Function]}
            />
            <button
              className="cubemap-remove"
              onClick={[Function]}
            >
              Remove
            </button>
          </div>
        </div>
      </div>
    </div>
  </article>
</div>
`;

exports[`redo_undo: ambientLight prepare first step: set currentSceneTreeNode test redo operate test redo two step undo step which from second to zero,redo step which from zero to second 1`] = `
<div
  className="header-item"
>
  <div
    className="component-item"
  >
    <span
      className="item-title"
      onClick={[Function]}
      onMouseOver={[Function]}
    >
      Setting
    </span>
  </div>
  <article
    className="wonder-settingScene-modal"
  >
    <div
      className="modal-item"
    >
      <div
        className="modal-item-header"
      >
        Scene
        <img
          onClick={[Function]}
          src="./public/img/close.png"
        />
      </div>
      <div
        className="modal-item-content"
      >
        <div
          className="setting-scene-ambient"
        >
          <div
            className="content-field"
          >
            <article
              className="color-pick"
            >
              <div
                className="item-header"
                title="set ambient color which is global."
              >
                AmbientColor
              </div>
              <div
                className="item-content item-color"
              >
                <div
                  className="color-hex"
                  onClick={[Function]}
                  style={
                    Object {
                      "background": "#1918e8",
                    }
                  }
                />
                <div
                  className="color-select"
                  onClick={[Function]}
                >
                  <img
                    src="./public/img/color.png"
                  />
                </div>
              </div>
            </article>
          </div>
        </div>
        <div
          className="setting-scene-skybox"
        >
          <div
            className="item-header"
            title="set scene skybox->cubemap"
          >
            Skybox Cubemap
          </div>
          <div
            className="item-content"
          >
            <div
              className="cubemap-img"
              onClick={[Function]}
            />
            <button
              className="cubemap-remove"
              onClick={[Function]}
            >
              Remove
            </button>
          </div>
        </div>
      </div>
    </div>
  </article>
</div>
`;

exports[`redo_undo: ambientLight prepare first step: set currentSceneTreeNode test undo operate test not undo 1`] = `
<div
  className="header-item"
>
  <div
    className="component-item"
  >
    <span
      className="item-title"
      onClick={[Function]}
      onMouseOver={[Function]}
    >
      Setting
    </span>
  </div>
  <article
    className="wonder-settingScene-modal"
  >
    <div
      className="modal-item"
    >
      <div
        className="modal-item-header"
      >
        Scene
        <img
          onClick={[Function]}
          src="./public/img/close.png"
        />
      </div>
      <div
        className="modal-item-content"
      >
        <div
          className="setting-scene-ambient"
        >
          <div
            className="content-field"
          >
            <article
              className="color-pick"
            >
              <div
                className="item-header"
                title="set ambient color which is global."
              >
                AmbientColor
              </div>
              <div
                className="item-content item-color"
              >
                <div
                  className="color-hex"
                  onClick={[Function]}
                  style={
                    Object {
                      "background": "#1918e8",
                    }
                  }
                />
                <div
                  className="color-select"
                  onClick={[Function]}
                >
                  <img
                    src="./public/img/color.png"
                  />
                </div>
              </div>
            </article>
          </div>
        </div>
        <div
          className="setting-scene-skybox"
        >
          <div
            className="item-header"
            title="set scene skybox->cubemap"
          >
            Skybox Cubemap
          </div>
          <div
            className="item-content"
          >
            <div
              className="cubemap-img"
              onClick={[Function]}
            />
            <button
              className="cubemap-remove"
              onClick={[Function]}
            >
              Remove
            </button>
          </div>
        </div>
      </div>
    </div>
  </article>
</div>
`;

exports[`redo_undo: ambientLight prepare first step: set currentSceneTreeNode test undo operate test undo one step step which from second to first 1`] = `
<div
  className="header-item"
>
  <div
    className="component-item"
  >
    <span
      className="item-title"
      onClick={[Function]}
      onMouseOver={[Function]}
    >
      Setting
    </span>
  </div>
  <article
    className="wonder-settingScene-modal"
  >
    <div
      className="modal-item"
    >
      <div
        className="modal-item-header"
      >
        Scene
        <img
          onClick={[Function]}
          src="./public/img/close.png"
        />
      </div>
      <div
        className="modal-item-content"
      >
        <div
          className="setting-scene-ambient"
        >
          <div
            className="content-field"
          >
            <article
              className="color-pick"
            >
              <div
                className="item-header"
                title="set ambient color which is global."
              >
                AmbientColor
              </div>
              <div
                className="item-content item-color"
              >
                <div
                  className="color-hex"
                  onClick={[Function]}
                  style={
                    Object {
                      "background": "#7df1e8",
                    }
                  }
                />
                <div
                  className="color-select"
                  onClick={[Function]}
                >
                  <img
                    src="./public/img/color.png"
                  />
                </div>
              </div>
            </article>
          </div>
        </div>
        <div
          className="setting-scene-skybox"
        >
          <div
            className="item-header"
            title="set scene skybox->cubemap"
          >
            Skybox Cubemap
          </div>
          <div
            className="item-content"
          >
            <div
              className="cubemap-img"
              onClick={[Function]}
            />
            <button
              className="cubemap-remove"
              onClick={[Function]}
            >
              Remove
            </button>
          </div>
        </div>
      </div>
    </div>
  </article>
</div>
`;

exports[`redo_undo: ambientLight prepare first step: set currentSceneTreeNode test undo operate test undo two step step which from second to zero 1`] = `
<div
  className="header-item"
>
  <div
    className="component-item"
  >
    <span
      className="item-title"
      onClick={[Function]}
      onMouseOver={[Function]}
    >
      Setting
    </span>
  </div>
  <article
    className="wonder-settingScene-modal"
  >
    <div
      className="modal-item"
    >
      <div
        className="modal-item-header"
      >
        Scene
        <img
          onClick={[Function]}
          src="./public/img/close.png"
        />
      </div>
      <div
        className="modal-item-content"
      >
        <div
          className="setting-scene-ambient"
        >
          <div
            className="content-field"
          >
            <article
              className="color-pick"
            >
              <div
                className="item-header"
                title="set ambient color which is global."
              >
                AmbientColor
              </div>
              <div
                className="item-content item-color"
              >
                <div
                  className="color-hex"
                  onClick={[Function]}
                  style={
                    Object {
                      "background": "#000000",
                    }
                  }
                />
                <div
                  className="color-select"
                  onClick={[Function]}
                >
                  <img
                    src="./public/img/color.png"
                  />
                </div>
              </div>
            </article>
          </div>
        </div>
        <div
          className="setting-scene-skybox"
        >
          <div
            className="item-header"
            title="set scene skybox->cubemap"
          >
            Skybox Cubemap
          </div>
          <div
            className="item-content"
          >
            <div
              className="cubemap-img"
              onClick={[Function]}
            />
            <button
              className="cubemap-remove"
              onClick={[Function]}
            >
              Remove
            </button>
          </div>
        </div>
      </div>
    </div>
  </article>
</div>
`;
